lib/index.ts

```diff
@@ -0,0 +1,405 @@
+
+/**
+ * Structure for recent bot statistics
+ */
+export interface RecentData {
+  time: string;
+  monthly_votes: number;
+  server_count: number;
+  shard_count: number;
+  total_votes: number;
+  monthly_votes_change: number;
+  monthly_votes_change_perc: number;
+  server_count_change: number;
+  shard_count_change: number;
+  total_votes_change: number;
+}
+
+export interface RecentDataRequest {
+  id: string;
+}
+
+export interface RecentDataResponse {
+  hourlyData: RecentData[];
+  dailyData: RecentData[];
+}
+
+/**
+ * Branded type for rankings limit (1-500)
+ */
+export type RankingsLimit = number & { readonly __brand: unique symbol };
+
+/**
+ * Validates if a number is a valid rankings limit
+ */
+export function isValidRankingsLimit(value: number): value is RankingsLimit {
+  return Number.isInteger(value) && value >= 1 && value <= 500;
+}
+
+export const DEFAULT_RANKINGS_LIMIT = 100 as RankingsLimit;
+
+export type RankingsSortBy =
+  | "monthly_votes_rank"
+  | "total_votes_rank"
+  | "server_count_rank"
+  | "shard_count_rank";
+
+export type RankingsSortMethod = "asc" | "desc";
+
+export interface RankingsRequest {
+  limit?: RankingsLimit;
+  sortBy: RankingsSortBy;
+  sortMethod: RankingsSortMethod;
+}
+
+export interface RankingsData {
+  id: string;
+  name: string;
+  monthly_votes: number;
+  monthly_votes_rank: number;
+  server_count: number;
+  server_count_rank: number;
+  total_votes: number;
+  total_votes_rank: number;
+  shard_count: number;
+  shard_count_rank: number;
+  monthly_votes_rank_change: number;
+  server_count_rank_change: number;
+  total_votes_rank_change: number;
+  shard_count_rank_change: number;
+}
+
+export interface RankingsResponse {
+  totalBotCount: number;
+  data: RankingsData[];
+}
+
+/**
+ * Custom error classes for better error handling
+ */
+export class TopStatsError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "TopStatsError";
+    Object.setPrototypeOf(this, TopStatsError.prototype);
+  }
+}
+
+export class RateLimitError extends TopStatsError {
+  constructor(
+    message: string,
+    public expiresIn: string,
+  ) {
+    super(message);
+    this.name = "RateLimitError";
+    Object.setPrototypeOf(this, RateLimitError.prototype);
+  }
+}
+
+export interface ClientOptions {
+  /** API Token for authentication */
+  token: string;
+  /** Optional custom base URL */
+  baseUrl?: string;
+}
+
+/**
+ * Main client class for interacting with the TopStats API
+ * @example
+ * ```typescript
+ * const client = new Client({ token: 'your-token' });
+ * // or
+ * const client = new Client('your-token');
+ *
+ * // Get bot info
+ * const bot = await client.getBot('123456789');
+ * ```
+ */
+export class Client {
+  private readonly token: string;
+  private readonly BASE_URL: string;
+
+  /**
+   * Creates a new TopStats API client
+   * @param options - Token string or client options object
+   */
+  constructor(options: string | ClientOptions) {
```

Comment:
It may be pertinent to make sure that the token value exists, otherwise throw an Error / console.warn

Reply:
fixed